### DISCUSSION TITLE
Issue with lighting

### DISCUSSION THREAD

**Discussion by Methew5** [sceneview/sceneform-android#261] [Q&A]
Hi.
I tried to use two options of LightEstimationConfig: Spectacular & Disabled.
In two cases I faced a problem:
1)Spectacular: when moving away, the model becomes darker
https://user-images.githubusercontent.com/69910337/145591863-d3eeb7fc-9d99-4814-b3b1-7219164da1a9.mp4
2)Disabled: on different walls, I have different colors of the model
https://user-images.githubusercontent.com/69910337/145591838-2c8350df-48ea-449b-839a-d372768a516e.mp4
@ThomasGorisse, or someone else, do you know, how can I fix that? Is it possible?

**Comment by ThomasGorisse** [comment]
Hi @Methew5 ,

Welcome to the light estimation world.

> 1)Spectacular: when moving away, the model becomes darker
> https://user-images.githubusercontent.com/69910337/145591863-d3eeb7fc-9d99-4814-b3b1-7219164da1a9.mp4

This is a "normal" behavior since your room seems very dark.
ARCore is updating the estimation while you go backward from the wall.
It first (when only seeing a white wall and nothing else) think it's a quite lighted room but estimate it more precisely when you give it more infos about the real environment (backing ward from the wall and seeing more dark things around)
LightEstimationConfig.SPECTACULAR vs LightEstimationConfig.REALISTIC mostly differs on the reflections parts and you will mainly only see differences if your model has more metallic than roughness material values.
Adjust the based reference/factored lighting values with:
```kotlin
sceneView.mainLight.intensity
```
and
```kotlin
sceneView.environment.indirectLight.intensity
```

> 2)Disabled: on different walls, I have different colors of the model
> https://user-images.githubusercontent.com/69910337/145591838-2c8350df-48ea-449b-839a-d372768a516e.mp4

This can be due to 2 things I think:
- The main directional light introduced with LightEstimation is also available in a LightEstimation.DISABLED context but kept fixed at the default Filament location/orientation and default indoor standard intensity as a normal non AR usage.
https://github.com/SceneView/sceneform-android/blob/f05ef5b21a98bca7b096a2dabdc08d8259b51ca1/core/src/main/java/com/gorisse/thomas/sceneform/light/Light.kt#L8-L11
So you can try to reduce it and the `sceneView.environment.indirectLight` intensity or call `sceneView.mainLight = null` if you want to completely disconnect what you are trying to render with the scene lighting but your planes will be less visible.
- You are maybe using the `baseColorFactor` instead of the `emissiveColorFactor` which may more correspond to what you are trying to achieve. Have a look at the [cursor.blend](https://github.com/SceneView/sceneview-android/blob/3340ec1dc7029aaeb4aeedd128a2465533f6e4a6/assets/models/cursor.blend) used by SceneView to inspect what material you should define to have a light independent color.

That being said, you can act on multiple things moreover then just modifying the LightEstimationConfig mode:

https://github.com/SceneView/sceneform-android/blob/c4ed1ba1b717ec3142e1335f623cc80cf838466f/core/src/main/java/com/gorisse/thomas/sceneform/light/LightEstimation.kt#L39-L96

Finally, have a look at the sample app for adjusting your desired values: 
[![Get it on Google Play](https://upload.wikimedia.org/wikipedia/commons/thumb/7/78/Google_Play_Store_badge_EN.svg/320px-Google_Play_Store_badge_EN.svg.png)](https://play.google.com/store/apps/details?id=com.gorisse.thomas.ar.environmentlights)

**Comment by Methew5** [comment]
@ThomasGorisse, thank you for your advice and time.
First of all, I consider the Spectacular config.
I have already tried all HDR options, created different custom configs, but it didn't help. 
Increasing indirect light intensity leads to a very bright model when the room is lit well.
Changing the main light intensity gives nothing.
But overall, the Spectacular config gives good results.
Thank you very much again, I really appreciate your work.
